common/plugin.hpp

```diff
@@ -24,6 +24,9 @@ namespace ILLIXR {
 
 		/**
 		 * @brief A method which Spindle calls when it starts the component.
+		 *
+		 * This is necessary because a constructor can't call derived virtual methods (due to
```

Comment:
This does not pertain to the logging call; It pertains to `threadloop`s, since the base class wants to call `_p_iteration`, which is virtual in the child class.

Reply:
I'm confused. Can you explain a bit more? What's the relation between `start()` and `_p_one_iteration()`?

Reply:
One might think of moving that the constructor, but `plugin::plugin()` calls `threadloop::threadloop()` which calls `the_plugin::the_plugin()` at the end. So creating a thread and calling `the_plugin::_p_one_iteration()` in `threadloop::threadloop()` is not safe; `the_plugin` is not constructed until the end of `threadloop::threadloop()`.

This is why we need `plugin::start()` and `threadloop::start()`. `plugin::start()` calls `threadloop::start()` which starts a thread to call `the_plugin::_p_one_iteration()`, after `the_plugin` is constructed. This is recommended in the [ISO C++ FAQ](https://isocpp.org/wiki/faq/strange-inheritance#calling-virtuals-from-ctor-idiom) when you would need to call a virtual method from the constructor.

Reply:
Thanks, that makes sense. But this is not clear from the current comment and needs to be documented better, so would be good to expand the comment.